python/ml4ir/base/features/preprocessing.py

```diff
@@ -37,7 +32,8 @@ def pop_fn(self, key):
 
 
 @tf.function
-def preprocess_text(feature_tensor, remove_punctuation=False, to_lower=False):
+def preprocess_text(feature_tensor, remove_punctuation=False, to_lower=False,
+                    punctuation=string.punctuation, replace_with_whitespace=False):
```

Comment:
I chose to have 'replace_with_whitespace' instead of specifying your own replacement because it is simpler. With your own replacement, I would have to go through some length to also strip leading and trailing such replacement (which would not be so straightforward to understand from an api perspective).